src/GraphQL/TaskExtensions.cs

```diff
@@ -42,14 +42,14 @@ internal static object GetResult(this Task task)
             }
         }
 
-        internal static async Task<IEnumerable<T>> WhereAsync<T>(this IEnumerable<T> items, Func<T, Task<bool>> predicate)
+        internal static async Task<IEnumerable<T>> WhereAsync<T, TState>(this IEnumerable<T> items, Func<T, TState, Task<bool>> predicate, TState state)
```

Comment:
I will gradually remove this method altogether, but for now I have added a parameter to it to transfer the state so that there are fewer closures.